contributors/design-proposals/storage/volume-provisioning.md

```diff
@@ -451,6 +461,38 @@ will be extended to contain StorageClass.Parameters.
 
 The existing provisioner implementations will be modified to accept the StorageClass configuration object.
 
+### External provisioner interface changes
+
+Sometimes PersistentVolume, PersistentVolumeClaim or StorageClass objects get
+new fields and/or different behavior is expected from an external provisioner.
+Old provisioner that don't know about these new fields must not provision PVCs
+that expect the provisioner knows about them.
+
+As an example, in 1.8 we introduced `StorageClass.MountOptions` and
```

Comment:
These two new 1.8 options are optional without defautling right?

Reply:
storageclass.reclaimpolicy will default to Delete (exact same behaviour as before when Delete was the only option)

Reply:
If the new fields are defaulted, then this scheme doesn't work because we will always have to set version to 1.8 since the new field is always set?

Reply:
Kubernetes can always check that 1.8 feature is really needed or 1.7-ish provisioner can do the job:

```go
if sc.MountOptions == "" && sc.ReclaimPolicy == "Delete" {
   /* use 1.7 */
} else {
  /* require 1.8 */
}
```